astroid/context.py

```diff
@@ -72,7 +76,7 @@ def __init__(self, path=None, nodes_inferred=None):
 
         e.g. the bound node of object.__new__(cls) is the object node
         """
-        self.extra_context = {}
+        self.extra_context: dict[NodeNG, InferenceContext] = {}
```

Comment:
Have you checked that this can't be `Instance` or its parent `Proxy`? In general, docstring typing is incorrect 😓

Reply:
Yes, I will cut this one.

Reply:
Changed it to `Any` instead.